tag.go

```diff
@@ -0,0 +1,87 @@
+package scale
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// GetMaxElementsFromValue returns the max number of elements for the specified field in
+// the struct passed as the v argument based on the 'scale' tag. It returns an error if v
+// is not a structure, if max is not specified for the field, the field doesn't exist or
+// there's a problem parsing the tag.
+func GetMaxElementsFromValue(v any, fieldName string) (uint32, error) {
```

Comment:
The `Get` prefix for a function/method is usually redundant.

```suggestion
func MaxScaleElementsForField(v any, fieldName string) (uint32, error) {
```